builtin/providers/datadog/resource_datadog_user_test.go

```diff
@@ -0,0 +1,230 @@
+package datadog
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"gopkg.in/zorkian/go-datadog-api.v2"
+)
+
+func TestAccDatadogUser_Required(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogUserDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckDatadogUserConfigRequired,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserExists("datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "handle", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_Email(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogUserDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckDatadogUserConfigEmail,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserExists("datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", "email@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "handle", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_IsAdmin(t *testing.T) {
```

Comment:
Same as previous remark, I'd consider collapsing `TestAccDatadogUser_Required`, `TestAccDatadogUser_Email` and `TestAccDatadogUser_IsAdmin`.